src/quantum_operator/operator.jl

```diff
@@ -1,25 +1,84 @@
+abstract type AbstractQuantumOperator end
+
+struct FermiCreation <: AbstractQuantumOperator end
+struct FermiAnnihilation <: AbstractQuantumOperator end
+struct Majorana <: AbstractQuantumOperator end
+struct BosonCreation <: AbstractQuantumOperator end
+struct BosonAnnihilation <: AbstractQuantumOperator end
+struct Classic <: AbstractQuantumOperator end
+
+Base.show(io::IO, ::Type{FermiCreation}) = print(io, "f⁺")
+Base.show(io::IO, ::Type{FermiAnnihilation}) = print(io, "f⁻")
+Base.show(io::IO, ::Type{Majorana}) = print(io, "f")
+Base.show(io::IO, ::Type{BosonCreation}) = print(io, "b⁺")
+Base.show(io::IO, ::Type{BosonAnnihilation}) = print(io, "b⁻")
+Base.show(io::IO, ::Type{Classic}) = print(io, "ϕ")
+
+isfermionic(::Type{AbstractQuantumOperator}) = error("not implemented!")
```

Comment:
I don't think it's necessary to implement isxxx() for all existing types. This could be done by implement default function isxxx(::Type{AbstractXXX))=false and implement the specific ones only when it returns true

Reply:
But then what if the user implements an operator they want to be Fermionic and forget to define isfermionic(::Type{MyFermionicOperator})? It is probably better to see an explicit error here rather than an implicit bug.

Reply:
BTW, if these properties are heavily used and requires extension, maybe it's a good idea to declare traits for them; if not, traits might be too complicated.
Something like

abstract type Statistics end
struct Bose <: Statistics end
struct Fermi <: Statistics end
Statistics(::Type{FermiCreation})=Fermi()
isfermionic(::Type{T}) where {T}=Statistics(T) isa Fermi

Reply:
> But then what if the user implements an operator they want to be fermionic and forget to define isfermionic(::Type{MyFermionicOperator})? It is probably better to see an explicit error here rather than an implicit bug.

Yes you're right. But it's always good to avoid duplicated codes as much as possible. I would suggest using either traits, or simply replace current duplicated code with something like
isfermionic(::Type{Union{F1,F2,F3...}}) = true
which is an abbreviation of current code.

Reply:
Agreed

Reply:
you have to do isfermionic(::Type{F}) where {F<Union{F1, F2, F3...}) = true